firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -91,9 +91,12 @@ public static Object convertToPlainJavaTypes(Object object) {
    *
    * @param object The representation of the JSON data
    * @param clazz The class of the object to convert to
+   * @param documentId The value to set to {@link com.google.firebase.firestore.DocumentId}
+   *     annotated fields in the custom class.
    * @return The POJO object.
    */
-  public static <T> T convertToCustomClass(Object object, Class<T> clazz) {
+  public static <T> T convertToCustomClass(Object object, Class<T> clazz, String documentId) {
```

Comment:
This parameter should be a `DocumentReference` or at least a `DocumentKey`. Just the string allows for

```java
@DocumentId
public String justTheLastPathComponent;
```
But we also want to support
```java
@DocumentId
public DocumentReference fullPath;
```

Reply:
Ah, should have fixed it. Done now.

Reply:
FYI- It doesn't look like you've pushed the fix yet, but you also haven't assigned it back to me / wilhuff, so perhaps you know that. :-)  Anyway, please assign it back when it's ready for another look!